config/jobs/periodic/knative/operator/main/operator-main.gen.yaml

```diff
@@ -34,6 +34,16 @@ periodics:
               TIMESTAMP=$(date +%s)
               K8S_BUILD_VERSION=$(curl https://storage.googleapis.com/k8s-release-dev/ci/latest.txt)
 
+              cleanup() {
```

Comment:
@Rajalakshmi-Girish can you check why it is not getting cleaned when it failed?

Reply:
@mkumatag  
In cases like [this](https://s3.us-south.cloud-object-storage.appdomain.cloud/ppc64le-prow-logs/logs/knative-operator-main-periodic/1854317866212397056/build-log.txt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=d5b0d1e12bb74ec5b0cca25bec3e0746%2F20241111%2Fus-south%2Fs3%2Faws4_request&X-Amz-Date=20241111T143935Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=725ba781c4333d285a5a96e55322aafbb172b011ccaadb7db9ccbe1c8a5abd5c) , if the job fails after the cluster is created when the e2e tests run, any test failure stops it from reaching the kubetest2 --down phase to destroy the VSIs, leaving resources running.

Reply:
We are handling this in our jobs in a different way, by disabling errexit while up/test runs and capturing $? after kubetest2 run.
 Cleanup/trap should be fine for this job.